feat: migrate`recreateClosed` to `recreateWhen` (#21039)
renovatebot/renovate: docs/usage/configuration-options.md

## recreateClosed

By default, Renovate detects if it proposed an update to a project before, and not propose the same one again.

> For example the Webpack 3.x case described above.

Reviewer: ```suggestion
For example the Webpack 3.x case described in the separateMajorMinor documentation.
```

## recreateClosed

By default, Renovate detects if it proposed an update to a project before, and not propose the same one again.
For example the Webpack 3.x case described above.
You can use `recreateClosed` to customize this behavior down to a per-package level.
For example we override it to `always` in the following cases where branch names and PR titles need to be reused:

- Package groups
- When pinning versions
- Lock file maintenance

There are total three possible options:

> 1. `always`: Recreates all closed/merged PRs

Reviewer: ```suggestion
1. `always`: Recreates all closed or blocking PRs
```

This could confuse people that Renovate will reopen normal merged PRs

## recreateClosed

By default, Renovate detects if it proposed an update to a project before, and not propose the same one again.

> For example the Webpack 3.x case described in the separateMajorMinor documentation.

Reviewer 2: ```suggestion
For example the Webpack 3.x case described in the `separateMajorMinor` documentation.
```

You can use `recreateClosed` to customize this behavior down to a per-package level.

> For example we override it to `always` in the following cases where branch names and PR titles need to be reused:

Reviewer 2: I'm not sure what `we override it to ...` refers to:

- we the maintainers override to `always` in our own personal Renovate bot config? 
- we the maintainers hard-coded some overrides into the Renovate bot code? (basically the bot ignores the presets/user config and always does `x`)?

Author: It means that if you use the options `auto | always` then still for immortal PRs the options we internally apply is `always`.

This sentence needs to change though, because now we have a third option `never` which doesn't let us apply `always` to immortal PRs

- Package groups
- When pinning versions
- Lock file maintenance

> There are total three possible options:

Reviewer 2: ```suggestion
You can select which behavior you want from Renovate bot:
```

How about this? Saying "There are total three possible options" is vague. Maybe a sentence like this one is clearer?

Author: Yes, I feel so too. Thanks♥

## recreateClosed

By default, Renovate detects if it proposed an update to a project before, and not propose the same one again.
For example the Webpack 3.x case described in the `separateMajorMinor` documentation.
You can use `recreateClosed` to customize this behavior down to a per-package level.
For example we override it to `always` in the following cases where branch names and PR titles need to be reused:

- Package groups
- When pinning versions
- Lock file maintenance

You can select which behavior you want from Renovate bot:

1. `always`: Recreates all closed or blocking PRs

> 2. `auto`: The default option. Recreates only immortal PRs

Reviewer: ```suggestion
2. `auto`: The default option. Recreates only immortal PRs (default)
```

Avoid setting `rebaseWhen=never` and then also setting `prCreation=not-pending` as this can prevent creation of PRs.

## recreateWhen

> By default, Renovate detects if it proposed an update to a project before, and will not propose the same update again.

Reviewer 2: ```suggestion
This feature used to be called `recreateClosed`.

By default, Renovate detects if it proposed an update to a project before, and will not propose the same update again.
```

Maybe we should add a note here, so that users searching for the old `recreateClosed` term can find the replacement?

> For example the Webpack 3.x case described in the `separateMajorMinor` documentation.

Reviewer 2: ```suggestion
For example the Webpack 3.x case described in the [`separateMajorMinor`](#separatemajorminor) documentation.
```

Let's link to that config option's heading.

You can use `recreateWhen` to customize this behavior down to a per-package level.
For example we override it to `always` in the following cases where branch names and PR titles must be reused:

- Package groups
- When pinning versions
- Lock file maintenance

> You can select which behavior you want from Renovate bot:

Reviewer 2: ```suggestion
You can select which behavior you want from Renovate:
```
